java/gradle/library.gradle

```diff
@@ -85,44 +31,20 @@ task javadocJar(type: Jar, dependsOn: javadoc) {
     destinationDir = new File("${libDir}")
 }
 
-artifacts {
-    archives jar, jarSources, javadocJar
-}
-
-signing {
-    required {
-        gradle.taskGraph.hasTask("signArchives") ||
-        gradle.taskGraph.hasTask("uploadArchives") ||
-        ( project.hasProperty('signing.keyId') &&
-          project.hasProperty('signing.password') &&
-          project.hasProperty('signing.secretKeyRingFile') )
-    }
-    sign configurations.archives
+// The maven plugin change in Gradle 7.3, for compatibility with older gradle versions
+// we split maven related tasks to separate files that use the corresponding builtin gradle
+// plug-ins
+project.ext.pomName = "${libDir}/${project.name}-${project.version}.pom"
+if(GradleVersion.current() >= GradleVersion.version('7.3')) {
```

Comment:
Aren't we always using the local Gradle version we have specified? Or is this really more for the offline Debian builds?

If so I would just skip the publishing stuff if we can. We only need to publish with one version I think.

Reply:
Right that is for Debian packaging that needs to run offline and with the local Gradle executable.